controllers/defaultremediation_controller.go

```diff
@@ -61,3 +81,43 @@ func (r *DefaultRemediationReconciler) SetupWithManager(mgr ctrl.Manager) error
 		For(&appv1alpha1.DefaultRemediation{}).
 		Complete(r)
 }
+
+func (r *DefaultRemediationReconciler) getRemediation(ctx context.Context, req ctrl.Request) *appv1alpha1.DefaultRemediation {
+	remediation := new(appv1alpha1.DefaultRemediation)
+	key := client.ObjectKey{Name: req.Name, Namespace: req.Namespace}
+	if err := r.Client.Get(ctx, key, remediation); err != nil {
+		if !errors.IsNotFound(err) {
+			r.Log.Error(err, "error retrieving remediation %s in namespace %s: %v", req.Name, req.Namespace)
+		}
+		return nil
+	}
+	return remediation
+}
+
+func (r *DefaultRemediationReconciler) deleteMachine(ctx context.Context, machine *unstructured.Unstructured) error {
+	if err := r.Client.Delete(ctx, machine); err != nil {
+		if !errors.IsNotFound(err) {
+			r.Log.Error(err, "error deleting machine %s in namespace %s: %v", machine.GetName(), machine.GetNamespace())
+			return err
+		}
+		r.Log.Info("machine: %s in namespace: %s is deleted , but remediation still exist", machine.GetName(), machine.GetNamespace(), err)
+	}
+	return nil
+}
+
+func getMachineRemediation(remediation *appv1alpha1.DefaultRemediation) *v1.OwnerReference {
+	for _, ownerRef := range remediation.OwnerReferences {
+		if ownerRef.Kind == machineKind {
+			return &ownerRef
+		}
+	}
+	return nil
+}
+func buildMachine(ref *v1.OwnerReference, remediation *appv1alpha1.DefaultRemediation) *unstructured.Unstructured {
+	machine := new(unstructured.Unstructured)
+	machine.SetName(remediation.Name)
+	machine.SetNamespace(remediation.Namespace)
+	machine.SetKind(machineKind)
```

Comment:
since this method will be called only if `ref.kind == machineKind` it is logically equal.
IMHO I think it's better to stick with the constant `machineKind` which is a bit simpler.